protocols/kad/src/addresses.rs

```diff
@@ -67,7 +67,7 @@ impl Addresses {
     /// otherwise unreachable.
     #[allow(clippy::result_unit_err)]
```

Comment:
I can do it if you want. That's no problem for me but it will probably be more annoying to release. I'll do what you decide but I think an `enum` would indeed be the best implementation.

Reply:
Yeah lets ship this fix first. We can add an issue to the next milestone to improve this :)